File: go/isogram/isogram.go
package isogram

import "strings"

// IsIsogram determines if a word has a repeating char.
func IsIsogram(s string) bool {
	charMap := map[string]int{}

	for i := range s {
		newChar := strings.ToLower(string(s[i]))
		if newChar != "-" && newChar != " " {
			if charMap[newChar] != 0 {
				return false
			}
			charMap[newChar] = 1
		}
	}

	return true
}

File: csharp/isogram/Isogram.cs
using System;
using System.Collections.Generic;
using System.Text.RegularExpressions;

public static class Isogram
{
    public static bool IsIsogram(string word)
    {
        HashSet<char> letters = new HashSet<char>();
        string normalized = new Regex("[- ]").Replace(word, "").ToLower();
        foreach(char c in normalized) {
            if (letters.Contains(c)) {
                return false;
            }
            letters.Add(c);
        }
        return true;
    }
}
